src/compiler/core.ts

```diff
@@ -1489,37 +1483,142 @@ namespace ts {
+        }
+
+        // Otherwise, fall back to ordinal comparison:
+        return createOrdinalStringCollator();
+    }
+
+    const caseInsensitiveCollator = createCaseInsensitiveStringCollator();
+
+    /**
+     * Performs a case-insensitive comparison between two strings.
+     *
+     * If supported by the host, the default locale is used for comparisons. Otherwise, an ordinal
+     * comparison is used.
+     */
+    export function compareStringsCaseInsensitive(a: string | undefined, b: string | undefined) {
         if (a === b) return Comparison.EqualTo;
         if (a === undefined) return Comparison.LessThan;
         if (b === undefined) return Comparison.GreaterThan;
-        if (ignoreCase) {
-            // Checking if "collator exists indicates that Intl is available.
-            // We still have to check if "collator.compare" is correct. If it is not, use "String.localeComapre"
-            if (collator) {
-                const result = localeCompareIsCorrect ?
-                    collator.compare(a, b) :
-                    a.localeCompare(b, /*locales*/ undefined, { usage: "sort", sensitivity: "accent" });  // accent means a ≠ b, a ≠ á, a = A
-                return result < 0 ? Comparison.LessThan : result > 0 ? Comparison.GreaterThan : Comparison.EqualTo;
-            }
+        const result = caseInsensitiveCollator.compare(a, b);
+        return result < 0 ? Comparison.LessThan : result > 0 ? Comparison.GreaterThan : Comparison.EqualTo;
+    }
 
-            a = a.toUpperCase();
-            b = b.toUpperCase();
-            if (a === b) return Comparison.EqualTo;
-        }
+    /**
+     * Performs a case-sensitive comparison between two strings.
+     */
+    export function compareStringsCaseSensitive(a: string | undefined, b: string | undefined) {
+        return compareValues(a, b);
+    }
 
-        return a < b ? Comparison.LessThan : Comparison.GreaterThan;
+    export function compareStrings(a: string | undefined, b: string | undefined, ignoreCase?: boolean): Comparison {
+        return ignoreCase ? compareStringsCaseInsensitive(a, b) : compareStringsCaseSensitive(a, b);
     }
 
-    export function compareStringsCaseInsensitive(a: string, b: string) {
-        return compareStrings(a, b, /*ignoreCase*/ true);
+    /**
+     * Performs a case-insensitive equality comparison between two strings.
+     *
+     * If supported by the host, the default locale is used for comparisons. Otherwise, an ordinal
+     * comparison is used.
+     */
+    export function equateStringsCaseInsensitive(a: string | undefined, b: string | undefined) {
+        return a === b
+            || a !== undefined
+            && b !== undefined
+            && caseInsensitiveCollator.equate(a, b);
+    }
+
+    /**
+     * Performs a case-sensitive equality comparison between two strings.
+     */
+    export function equateStringsCaseSensitive(a: string | undefined, b: string | undefined) {

```

Comment:
It will be inlined, plus the const shows up as a variable and not a function in intellisense:

![equate](https://user-images.githubusercontent.com/3902892/32020194-398bf2bc-b984-11e7-8bfb-98914f2eee4d.png)

Not a big deal, but we have plenty of other cases where we write functions like this.